src/App.jsx

```diff
@@ -2,22 +2,42 @@ import { useState, useEffect } from 'react';
 import TodoList from './TodoList';
 import AddTodoForm from './AddTodoForm';
 
-function useSemiPersistentState() {
+function App() {
   const [todoList, setTodoList] = useState(() => {
     const savedList = localStorage.getItem("savedTodoList");
     return savedList ? JSON.parse(savedList) : [];
   });
 
+  const [isLoading, setIsLoading] = useState(true);
+  const [todoTitle, setTodoTitle] = useState('');
+
   useEffect(() => {
-    localStorage.setItem("savedTodoList", JSON.stringify(todoList));
-  }, [todoList]);
+    const fetchTodoList = new Promise((resolve, reject) => {
+      setTimeout(() => {
+        resolve({
+          data: {
+            todoList: []
+          }
+        });
+      }, 2000);
+    });
 
-  return [todoList, setTodoList];
-};
+    fetchTodoList.then((result) => {
+      setTodoList((prevTodoList) => prevTodoList.length ? prevTodoList : result.data.todoList);
```

Comment:
then you can now set the result here with 
```suggestion
setTodoList(result.data.todoList)
```